sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -401,6 +401,15 @@ abstract class NewRule {
      */
     public abstract NewRule setType(RuleType t);
 
+    /**
+     * Add a rule description section. The sections must be added in the right order.
+     * For backward compatibility, one of the old method {@link #setHtmlDescription(String)} or {@link #setHtmlDescription(URL)} still
+     * need to be called on top of that one.
+     *
+     * @since 9.5
+     */
+    public abstract NewRule addDescriptionSection(RuleDescriptionSection ruleDescriptionSection);
```

Comment:
We decided to not mark it as deprecated due to the following: first SonarQube will be updated to the new API, then SonarLint & SonarCloud.

Also, we don't control which version of the plugins will be executed on which version of SonarLint (it depends on which SQ server it is connected to).

As a result, the plugins will need to set during a transition period both the old API (`setHtmlDescription`) and the new one (`addDescriptionSection`), so it's expected that both methods are still used.

Maybe some javadoc should be added to make it clearer.

Reply:
We didn't want to deprecate it to not give to plugin-writers the wrong feeling that they should clean up their code and not call this method. We actually want the opposite: we want them to still use the method during the transition period.

As for "generating" a description, for instance by concatenating sections, I'm afraid it might generate problems in the future when the new API will support more features than the old one (for instance images, context, etc.). We could put in place a "backward-sanitization" mechanism; for instance removing images tags. But this could lead to inconsistent descriptions; for instance a textual reference to an image not shown.

Reply:
got it, thanks